python/pyspark/sql/tests/test_dataframe.py

```diff
@@ -884,7 +888,10 @@ def test_sample(self):
 
         self.assertRaises(TypeError, lambda: self.spark.range(1).sample(seed="abc"))
 
-        self.assertRaises(IllegalArgumentException, lambda: self.spark.range(1).sample(-1.0))
```

Comment:
@itholic should we and a new subclass for `IllegalArgumentException`?

Reply:
Gentle ping, @itholic .

Reply:
@zhengruifeng Maybe are you asking if we want to add a new subclass for `IllegalArgumentException` such as `PySparkIllegalArgumentException` ?

If so, I don't think so because the `IllegalArgumentException` already has its own proper error class and handled in JVM side.

Reply:
Ah, I got it. We should catch `IllegalArgumentException ` properly before `SparkConnectException` occurred in Spark Connect as mentioned in https://github.com/apache/spark/pull/39765/files#r1089673621.

Let me handle this in a separate ticket.